dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-to-payment-details.ts

```diff
@@ -0,0 +1,41 @@
+import { getFeeRecordPaymentEntityGroupReconciliationData, getFeeRecordPaymentEntityGroupStatus } from '../../../../../helpers';
+import { mapPaymentEntityToPayment } from '../../../../../mapping/payment-mapper';
+import { mapFeeRecordEntityToFeeRecord } from '../../../../../mapping/fee-record-mapper';
+import { FeeRecordPaymentEntityGroup } from '../../../../../types/fee-record-payment-entity-group';
+import { PaymentDetails } from '../../../../../types/utilisation-reports';
+
+/**
+ * Maps the fee record payment entity groups to the payment details
+ * @param paymentsWithFeeRecords - The fee record payment entity groups.
+ * These should be flattened such that there is at most one payment in each group.
+ * @returns The payment details
+ * @throws {Error} If the group has more than one payment
+ */
+export const mapToPaymentDetails = async (paymentsWithFeeRecords: FeeRecordPaymentEntityGroup[]): Promise<PaymentDetails[]> => {
+  const mappedPaymentDetails: PaymentDetails[] = [];
+
+  for (const group of paymentsWithFeeRecords) {
+    if (group.payments.length > 1) {
+      throw new Error('Error mapping payments to payment details - groups must have at most one payment.');
```

Comment:
```suggestion
      throw new Error('Error mapping group payments to payment details - group has more than one payment.');
```

Reply:
We don't at the minute - unfortunately we don't have any group or correlation ID from the point at which we get these groups in `getFeeRecordPaymentEntityGroups` - below:

https://github.com/UK-Export-Finance/dtfs2/blob/df928779cc08be8ce401bab871fdca921416af05/dtfs-central-api/src/helpers/get-fee-record-payment-entity-groups.ts#L18


I think the best logging we can add is logging the whole group just before we throw this error, with a similar error message - I've added this, open to comments on how you feel about this thanks

I've also made the update to the error message thanks

Reply:
Nice one thanks @sw-mattw 

I'm not sure about the object logging, but it's the only sensible thing we can log I think it's better than nothing